tests/thread_test.c

```diff
@@ -53,3 +53,37 @@ static int s_test_thread_creation_join_fn(struct aws_allocator *allocator, void
 }
 
 AWS_TEST_CASE(thread_creation_join_test, s_test_thread_creation_join_fn)
+
+static uint32_t s_atexit_call_count = 0;
+static void s_thread_atexit_fn(void *user_data) {
+    (void)user_data;
+    s_atexit_call_count = 1;
+}
+
+static void s_thread_atexit_fn2(void *user_data) {
+    (void)user_data;
+    s_atexit_call_count = 2;
```

Comment:
`AWS_FATAL_ASSERT(s_atexit_call_count == 1);`? Otherwise, if chaining breaks, and we just call the first fn, we're broke.

Reply:
You are full of good ideas today